bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java

```diff
@@ -62,8 +62,8 @@ public StationDataResponse getStationsData(@Nullable String deviceId, boolean ge
         return response;
     }
 
-    public NAMain getStationData(String deviceId) throws NetatmoException {
-        ListBodyResponse<NAMain> answer = getStationsData(deviceId, true).getBody();
+    public NAMain getStationData(String deviceId, boolean getFavorites) throws NetatmoException {
```

Comment:
In line 52:
`@param getFavorites Whether to include the user's favorite Weather Stations in addition to the user's own Weather Stations (optional, default to false)`

Does it make sense?

First, there is no "default" for this parameter as it's a required parameter. Perhaps the intention was the mention that if parameter is left out in API call, it will default to false. But as written here, it's not the API which is described, but the actual parameter.

Second, perhaps we should introduce a method overload which could actually default to false. Or even a different method, `getStationDataWithFavorites` for clarity.

Reply:
You're right, the comment on the parameter is certainly more a comment for the API.
The binding systematically provides a value.
So the parameter description should be fixed.
No need for another method IMHO.
Just to clarify for you, either you want the data from your own station, in this case you have to provide the device_id and get_favorites can be omitted or can be set to false; setting it to true will be ok bu will return also some data you don't need. If you want the data from a favorite station, you have to provide the device_id and get_favorites must be set to true.

Reply:
So from your explanation, if I understand it correctly, perhaps something like this would be be easier to understand and read:

```
public NAMain getStationData(String deviceId) throws NetatmoException {
    return getStationData(deviceId, false);
}

public NAMain getGuestStationData(String deviceId) throws NetatmoException {
    return getStationData(deviceId, true);
}

private NAMain getStationData(String deviceId, boolean getFavorites) throws NetatmoException {
    ...
}
```

This would hide the complexity and avoid the combination of providing **own deviceId** and **getFavorites = true**, which will fetch unneeded data.

There is also this call from `NetatmoDiscoveryService` to be considered:
`getStationsData(null, true)`

Reply:
Ok, why not

Reply:
`getStationsData` (mind the plural Stations) is only called externally from `NetatmoDiscoveryService` and always with deviceId null, so it would make sense to expose an explicit method for this case, e.g.:

```
public StationDataResponse getStationsData()
```

Reply:
There might also be a difference between favorite stations and guest stations, which I'm not aware of, so please keep this in mind also with the naming - if refactoring into something which will be more understandable at first sight.

Reply:
I just pushed changes but without a dedicated method for discovery. I am going to add it.

Reply:
Done. You can have a final look.